### PR TITLE
[Docs] `subtractXYZ`

### LINKED ISSUE #2017: Document `subtractXYZ` functions.
Create documentation for the `subtractXYZ` functions. See #1833 for details on how to complete this issue.

- [ ] subtractDays
- [ ] subtractHours
- [ ] subtractInterval
- [ ] subtractMicroseconds
- [ ] subtractMilliseconds
- [ ] subtractMinutes
- [ ] subtractMonths
- [ ] subtractNanoseconds
- [ ] subtractQuarters
- [ ] subtractSeconds
- [ ] subtractTupleOfIntervals
- [ ] subtractWeeks
- [ ] subtractYears

Child issue of #1833.

### PR DESCRIPTION
**Pull request by Blargian**
Closes [#2017](https://github.com/ClickHouse/clickhouse-docs/issues/2017) as part of the [project](https://github.com/ClickHouse/clickhouse-docs/issues/1833) document missing functions and update ones which are insufficiently documented. 

This PR makes updates / adds where missing the following functions:

-  `subtractDays`
-  `subtractHours`
-  `subtractInterval`
-  `subtractMicroseconds`
-  `subtractMilliseconds`
-  `subtractMinutes`
-  `subtractMonths`
-  `subtractNanoseconds`
-  `subtractQuarters`
-  `subtractSeconds`
-  `subtractTupleOfIntervals`
-  `subtractWeeks`
-  `subtractYears`

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 33ec14758de2c452ec7ef38bbf4152d55843ece5 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63251/33ec14758de2c452ec7ef38bbf4152d55843ece5/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by nikitamikhaylov — APPROVED** [review]

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 1]
```diff
@@ -2580,13 +2580,27 @@ SELECT
...
 
-``` sql
+```sql
+subtractYears(date, x)
+```
+
+**Parameters**
+
+- `date`: Date / date with time to subtract specified number of years from. [date](../data-types/date.md)/[date32](../data-types/date32.md)/[datetime](../data-types/datetime.md)/[datetime64](../data-types/datetime64.md), [String](../data-types/string.md).
+- `x`: Number of years to subtract. [(U)Int*](../data-types/int-uint.md), [float*](../data-types/float.md).
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/date-time-functions.md:2596]: `n` or `num` as variable name would be more idomatic

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 2]
```diff
@@ -2597,12 +2611,456 @@ SELECT
...
+
+**Syntax**
+
+```sql
+subtractQuarters(date, x)
+```
+
+**Parameters**
+
+- `date`: Date / date with time to subtract specified number of quarters from. [date](../data-types/date.md)/[date32](../data-types/date32.md)/[datetime](../data-types/datetime.md)/[datetime64](../data-types/datetime64.md), [String](../data-types/string.md).
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/date-time-functions.md:2632]: `date` --> `Date`
`date32` --> `Date32`
`datetime` --> `DateTime`
`datetime64` --> `DateTime64`

**Comment by Blargian** [comment]
@rschu1ze opened PR #63418 to fix above mentioned issues.

**Pull request merged**